pkg/i2gw/aggregator.go

```diff
@@ -284,34 +295,56 @@ func (rg *ingressRuleGroup) toHTTPRoute() (gatewayv1beta1.HTTPRoute, []error) {
 			Matches: []gatewayv1beta1.HTTPRouteMatch{*match},
 		}
 
-		var numWeightedBackends, totalWeightSet int32
-		for _, path := range paths {
-			backendRef, err := toBackendRef(path.path.Backend)
-			if err != nil {
-				errors = append(errors, err)
-				continue
-			}
-			if path.extra != nil && path.extra.canary != nil && path.extra.canary.weight != 0 {
-				weight := int32(path.extra.canary.weight)
-				backendRef.Weight = &weight
-				totalWeightSet += weight
-				numWeightedBackends++
+		backendRefs, errs := rg.calculateBackendRefWeight(paths)
+		errors = append(errors, errs...)
+		hrRule.BackendRefs = backendRefs
+
+		httpRoute.Spec.Rules = append(httpRoute.Spec.Rules, hrRule)
+	}
+
+	return httpRoute, errors
+}
+
+func (rg *ingressRuleGroup) calculateBackendRefWeight(paths []ingressPath) ([]gatewayv1beta1.HTTPBackendRef, field.ErrorList) {
+	var errors field.ErrorList
+	var backendRefs []gatewayv1beta1.HTTPBackendRef
+
+	var numWeightedBackends, totalWeightSet int32
+	var weightTotal = 100
```

Comment:
Worth adding a comment that this is related to ingress nginx and it is the default for `nginx.ingress.kubernetes.io/canary-weight-total` based on https://github.com/kubernetes/ingress-nginx/blob/main/docs/user-guide/nginx-configuration/annotations.md#canary

Reply:
Good idea, I'll add the comment.
Anyway, I think we should remove the support for nginx annotations from here and handle it separately with the other provider-specific logic, as you suggested.